File: be/src/main/java/com/netcracker/edu/backend/service/ProjectService.java
package com.netcracker.edu.backend.service;

import com.netcracker.edu.backend.entity.Project;
import com.netcracker.edu.backend.entity.User;

public interface ProjectService {
    Project saveProject(Project project);
    Iterable<Project>getProjects(int page);
    Iterable<Project>getProjectsAll();
    Project loadByCode(String name);
}

File: frontend/src/app/service/project/project.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Project} from "../../model/project";

@Injectable({
  providedIn: 'root'
})
export class ProjectService {
  currentProject:Project;

  constructor(private http: HttpClient) {}
  saveProject(project:Project):Observable<Project>{
    return this.http.post<Project>('/api/project',project);
}
  getProjects(page:number): Observable<Project[]> {
    return this.http.get<Project[]>('/api/project/page?page='+page);
  }
  getProjectsAll  (): Observable<Project[]> {
    return this.http.get<Project[]>('http://localhost:8080/api/project/all');
  }
}
